src/Magnum/EigenIntegration/Integration.h

```diff
@@ -369,6 +370,51 @@ template<class To, std::size_t size, class T> inline To cast(const Math::Vector<
+    using MatrixT = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic, Eigen::RowMajor>;
+    using StrideT = Eigen::Stride<Eigen::Dynamic, Eigen::Dynamic>;
+    auto size = from.size();
+    auto stride = from.stride();
+    return Eigen::Map<MatrixT, Eigen::Unaligned, StrideT>(reinterpret_cast<T*>(from.data()), size[0], size[1], StrideT(stride[0]/sizeof(T), stride[1]/sizeof(T)));
+}
+
+/**
+@brief Convert Corrades StridedArrayView1D to Eigens Dynamic Vector Type.
+Since for a one dimensional StridedArrayView there is no column or row version
+we always return an Eigen Column Vector.
+*/
+template<class T> inline Eigen::Map<Eigen::Matrix<T, Eigen::Dynamic, 1>, Eigen::Unaligned, Eigen::InnerStride<>> arrayCast(const Containers::StridedArrayView1D<T>& from) {
+    using MatrixT = Eigen::Matrix<T, Eigen::Dynamic, 1>;
+    return Eigen::Map<MatrixT, Eigen::Unaligned, Eigen::InnerStride<>>(reinterpret_cast<T*>(from.data()), from.size(), Eigen::InnerStride<>(from.stride()/sizeof(T)));
+}
+
+/**
+@brief Convert an Eigen Expression to Corrades StridedArrayView1D
+If it is known at compile time that the Eigen Expression has either
+one Column or one Row, then this function maps the Eigen Expression
+to a one dimensional StridedArrayView.
+*/
+template<class Derived> inline typename std::enable_if<(Eigen::internal::traits<Derived>::ColsAtCompileTime == 1) || (Eigen::internal::traits<Derived>::RowsAtCompileTime == 1), Containers::StridedArrayView1D<typename Derived::Scalar>>::type arrayCast(const Eigen::DenseCoeffsBase<Derived, Eigen::DirectWriteAccessors>& from) {
+    using Scalar = typename Derived::Scalar;
+    return Containers::StridedArrayView1D<Scalar>({const_cast<Scalar*>(&from(0,0)), std::size_t(from.size()*from.innerStride())}, std::size_t(from.size()), std::size_t(from.innerStride())*sizeof(Scalar));
+}
+
+/**
+@brief Convert an Eigen Expression to Corrades StridedArrayView2D
+Overload that takes care of any Eigen Expression that was not handled by @ref Magnum::EigenIntegration::arrayCast().
+@snippet EigenIntegration.cpp IntegrationStridedArrayView
+*/
+template<class Derived> inline typename std::enable_if<(Eigen::internal::traits<Derived>::ColsAtCompileTime == Eigen::Dynamic) && (Eigen::internal::traits<Derived>::RowsAtCompileTime == Eigen::Dynamic), Containers::StridedArrayView2D<typename Derived::Scalar>>::type arrayCast(const Eigen::DenseCoeffsBase<Derived, Eigen::DirectWriteAccessors>& from) {
+    using Scalar = typename Derived::Scalar;
+    /* we assume that the Eigen expression is a slice of some continuous piece of memory */
+    std::size_t size = from.rows()*from.colStride() + from.cols()*from.rowStride();
+    return Containers::StridedArrayView2D<Scalar>({const_cast<Scalar*>(&from(0,0)), std::size_t(size)}, {std::size_t(from.rows()), std::size_t(from.cols())}, {Long(from.rowStride())*Long(sizeof(Scalar)), Long(from.colStride())*Long(sizeof(Scalar))});
```

Comment:
All `Long`s should be `std::ptrdiff_t` here, otherwise you'd get warnings/errors on 32bit.